exercise_perseus_render/assets/src/vue/index.vue

```diff
@@ -1,23 +1,59 @@
 <template>
 
   <div id="exercise-container">
-    <perseus v-if="item" :item="item"></perseus>
+    <assessment-wrapper
+      v-ref:wrapper
+      v-if="exercise"
+      :itemId="itemId"
+      :masteryModel="exercise.mastery_model"
+      :masterySpacingTime="exercise.masterySpacingTime">
+      <perseus v-if="item" :item="item"></perseus>
+    </assessment-wrapper>
   </div>
 
 </template>
 
 
 <script>
 
-  const defaultQuestion = require('./question.json');
-
   module.exports = {
     data: () => ({
-      item: defaultQuestion,
+      item: undefined,
+      exercise: undefined,
+      itemId: undefined,
     }),
+    methods: {
+      setItemId() {
+        const items = this.exercise.all_assessment_items;
+        const attempts = this.$refs.totalattempts;
+        this.itemId = items[attempts % items.length];
+        this.$emit('itemIdSet');
+      },
+      setItemData() {
+        this.Kolibri.client(
+          `${this.defaultFile.storage_url}/${this.itemId}.json`
+          ).then((itemResponse) => {
+            const itemData = itemResponse.entity.itemData.replace(
+              '${aronsfacehere}',
+              this.defaultFile.storage_url);
+            this.item = JSON.parse(itemData);
+          });
+      }
+    },
     components: {
       perseus: require('./perseus'),
+      assessmentWrapper: require('kolibri/core/vue/components/assessmentWrapper'),
```

Comment:
Might need to be fixed, yes. If only the person writing this was the person who had defined the API.